app/models/remote_execution_provider.rb

```diff
@@ -4,7 +4,15 @@ class RemoteExecutionProvider
 
   class << self
     def provider_for(type)
-      providers[type.to_s] || providers[:SSH]
+      providers[type.to_s] || providers[:script]
+    end
+
+    def registered_name
+      providers.find { |_name, klass| klass == self }.first
```

Comment:
This is bit hard to ready, could we use https://ruby-doc.org/core-2.7.0/Hash.html#method-i-key ?
```suggestion
      klass = self
      providers.key(klass)
```

Reply:
TIL